uberfire-project/uberfire-project-api/src/main/java/org/guvnor/common/services/project/model/GAV.java

```diff
@@ -34,7 +37,11 @@ public GAV() {
     }
 
     public GAV(final String gavString) {
-        String[] split = gavString.split(":");
+        String[] split = checkNotNull("gavString", gavString)
+                .split(":");
+        if (split.length != 3) {
```

Comment:
Fix ArrayIndexOutOfBounds, make NPE explicit..

Reply:
@jhrcek Looks like the coordinates can have more than three elements (e.g. scope, packaging) in some contexts. This code should probably allow this. Currently it causes a test failure in `kie-wb-distributions` but it may cause more problems on other places.

![projectlibraryintegrationtest_importandbuildprojectfromstockrepository](https://user-images.githubusercontent.com/673386/41097522-8415053c-6a58-11e8-841e-704bb33d883e.png)

```
at org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest.lambda$importAndBuildProjectFromStockRepository$0(ProjectLibraryIntegrationTest.java:52)
at org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest.importBuildDeployAndCheckArtifact(ProjectLibraryIntegrationTest.java:71)
at org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest.importAndBuildProjectFromStockRepository(ProjectLibraryIntegrationTest.java:49)
```

Reply:
@yurloc Thanks for reporting. That's what we get when people represent composite datatypes using primitives like String :P I'll push fix shortly..